src/utils/pyproject.ts

```diff
@@ -40,9 +48,21 @@ function parsePyproject(pyprojectContent: string): string | undefined {
   )
     .flat()
     .filter((item: string | object) => typeof item === "string");
-  return getRuffVersionFromAllDependencies(
+  const version = getRuffVersionFromAllDependencies(
     dependencies.concat(optionalDependencies, devDependencies),
   );
+  if (version) return version;
+
+  // Special handling for Poetry until it supports PEP 735
+  // See: <https://github.com/python-poetry/poetry/issues/9751>
+  const poetryGroups = Object.values(pyproject?.tool?.poetry?.group ?? {});
+  return poetryGroups
+    .flatMap((group) => Object.entries(group.dependencies))
+    .map(([name, spec]) => {
+      if (name === "ruff" && typeof spec === "string") return spec;
+      return undefined;
+    })
+    .find((version) => version !== undefined);
```

Comment:
```suggestion
  // Special handling for Poetry until it supports PEP 735
  // See: <https://github.com/python-poetry/poetry/issues/9751>
  const poetryDependencies = Object.values(pyproject?.tool?.poetry?.group || {})
    .flatMap((group) => Object.entries(group.dependencies))
    .filter((item: string | object) => typeof item === "string");
```
Have I missed something why we can't use the same style as for the other groups?

Reply:
The problem is that poetry groups looks like this:

```toml
tool.poetry.group.dev.dependencies]
ruff = "0.8.3"
```

The function expects something like `ruff==0.8.3`. First I thought about just mapping it to the expected format, but that felt a bit too hacky. Let me know if you want to take a different approach.

Reply:
And that is why we can't have nice things 🫤 
I will resolve this conversation and continue in a new review.